modules/route53/main.tf

```diff
@@ -0,0 +1,11 @@
+resource "aws_route53_zone" "zone" {
+  name = var.dns_zone
+}
+
+resource "aws_route53_record" "record" {
+  zone_id = aws_route53_zone.zone.zone_id
+  name    = var.dns_name
+  type    = var.dns_type
+  ttl     = var.dns_ttl
+  records = aws_route53_zone.zone.name_servers
```

Comment:
this should map to one of `ip_address`, `ipv6_address`, `alias` or `name` as supplied in resource inputs.